components/MachineForm/index.js

```diff
@@ -1,12 +1,28 @@
+import { useEffect } from "react";
 import {
   FormLabel,
   SubmitButton,
   FormInput,
   FormArea,
   StyledForm,
 } from "./MachineForm.Styled";
+import { useState } from "react";
+import Image from "next/image";
+
+export default function MachineForm({
+  onSubmit,
+  machine,
+  imageThere,
+  imageUrl,
+}) {
+  const [pictureDefault, setPictureDefault] = useState(
+    imageUrl || machine?.machineImage
+  );
+
+  useEffect(() => {
+    setPictureDefault(imageUrl || machine?.machineImage);
+  }, [imageUrl, machine?.machineImage]);
```

Comment:
Why is this state called `pictureDefault`? This is the image right? What is default about it?

I don't really get what the effect is supposed to do. Why does the state net to be reset?

Reply:
thats a good question indeed i got so tangled up at some point🫣 
and im sure there is a better way to solve it.
i think i used it to make the defaultValue of the type=url input of the MachineForm rendered on the edit page change the displayed url once an image was uploaded.
which got kind of obsolete once i added the image preview but i couldnt let go of it after sacrificing my brain cells for 🥸